ProcessLib/ThermoHydroMechanics/Tests.cmake

```diff
@@ -0,0 +1,86 @@
+# ThermoHydroMechanics; Small deformation, linear poroelastic, homogeneous
+AddTest(
+    NAME ThermoHydroMechanics_square_1e0
+    PATH ThermoHydroMechanics/Linear/Square_sealed_homogeneous
+    EXECUTABLE ogs
+    EXECUTABLE_ARGS square_1e0.prj
+    WRAPPER time
+    TESTER vtkdiff
+    REQUIREMENTS NOT OGS_USE_MPI
+    ABSTOL 1e-8 RELTOL 1e-8
+    DIFF_DATA
+    expected_square_1e0_pcs_0_ts_10_t_1000.000000.vtu square_1e0_pcs_0_ts_10_t_1000.000000.vtu displacement displacement
+    expected_square_1e0_pcs_0_ts_10_t_1000.000000.vtu square_1e0_pcs_0_ts_10_t_1000.000000.vtu pressure pressure
+    expected_square_1e0_pcs_0_ts_10_t_1000.000000.vtu square_1e0_pcs_0_ts_10_t_1000.000000.vtu temperature temperature
+    expected_square_1e0_pcs_0_ts_10_t_1000.000000.vtu square_1e0_pcs_0_ts_10_t_1000.000000.vtu epsilon_xx epsilon_xx
+    expected_square_1e0_pcs_0_ts_10_t_1000.000000.vtu square_1e0_pcs_0_ts_10_t_1000.000000.vtu epsilon_yy epsilon_yy
+)
+
+AddTest(
+    NAME ThermoHydroMechanics_square_1e0_analyitcal
+    PATH ThermoHydroMechanics/Linear/Square_sealed_homogeneous
+    EXECUTABLE ogs
+    EXECUTABLE_ARGS square_1e0.prj
+    WRAPPER time
+    TESTER vtkdiff
+    REQUIREMENTS NOT OGS_USE_MPI
+    ABSTOL 1e-3 RELTOL 1e-3
```

Comment:
This is rather coarse tolerance. How come?

Reply:
changed the tolerance to 1e-4

Reply:
For comparison with analytical solution, 1e-4 is already good for pressure.

Reply:
But the relative tolerance is also 1e-3 and the vtkdiff's final result is `(abs(err) < tol_a || rel(err) < tol_r)`.

Reply:
The analytical result is 0.104160002 MPa and numerical result is 0.104160000 MPa, the slightly difference is from the different treatment of solid and fluid density of numerical solution and analytical solution.

Reply:
Not knowing the details IMHO 0.104160002 MPa and 0.104160000 MPa are practically the same (rel. diff. 2e-8). But what do you mean by
> different treatment of solid and fluid density of numerical solution and analytical solution

Why is there a different treatment? Differences in the analytical and numerical setting make the whole reasoning about test cases unnecessarily difficult.

Reply:
The difference treatment is that numerical solution considers the density change for solid due to thermal expansion which the analytical solution does not. The influence is very small so the result is still very accurate.

Reply:
Maybe that's now too picky: But in general then IMHO it would be better to set the thermal expansion to zero in the simulation, too. Because strictly speaking you are solving two different problems numerically vs. analytically.

Reply:
I do like Chirstoph's suggestion.
Also while the results are ok, the much higher relative tolerance should be set in the ctest's settings.